src/Projection.jl

```diff
@@ -0,0 +1,82 @@
+
+abstract type AbstractProjection end
+
+struct Projection{C,M,K} <: AbstractProjection
+    contract::C
+    model::M
+    kind::K
+end
+
+Projection(c) = Projection(c, NullModel(), CashflowProjection())
+Projection(c, m) = Projection(c, m, CashflowProjection())
+
+function Transducers.asfoldable(c::C) where {C<:FinanceCore.AbstractContract}
+    Projection(c) |> Map(identity)
+end
+Base.collect(p::P) where {P<:AbstractProjection} = p |> Map(identity) |> collect
+Base.collect(c::C) where {C<:FinanceCore.AbstractContract} = Projection(c) |> Map(identity) |> collect
+
+
+# controls what gets produced from the model,
+# e.g. if you just want cashflows or you want full amortization schedule, etc
+abstract type ProjectionKind end
+
+struct CashflowProjection <: ProjectionKind end
+
+@inline function Transducers.__foldl__(rf, val, p::Projection{C,M,K}) where {C<:Cashflow,M,K}
+    for i in 1:1
```

Comment:
This confuses me a bit

Reply:
I just pushed an update to that file with a bunch of commentary, hopefully that helps?